### ISSUE TITLE
expand social media paste behavior for images to more content types: 5pts

### ISSUE THREAD

**Issue by paregorios** [isawnyu/isaw.web#457]
_This issue is **blocked** for completion of #472 so that we are using proper image scales everywhere with the `og:image` tag._

Our `event` and `news item` objects use a good image (but too small; see #472) when pasted into a post on twitter, facebook, and mastodon. If there is a lead/primary image, that's what gets used in the preview. But other content items don't seem to include the metadata the social media sites are expecting (`<meta property="og:image" ...`). Instead of the lead/primary or first image in the content item, a giant ISAW logo is used as the preview image. We'd like that behavior to change as follows.

Notes: For any of the below content types, if the recommended image field is empty, look for the `body` or the first rich content field and, if there is an embedded image therein, use it. Otherwise, do not recommend an image.

- A **collection:** use the `lead image`.
- An **exhibition object:** use the `lead image`.
- A **file:** use the `lead image`.
- A **folder**
   - Folder does not have a content item selected as "default view": use the **`lead image` on the folder.**
   - Folder has a content item selected as "default view": use the **`lead image` (or similar for weird content items in this list) on the default view content item** if present. If not, use the `lead image` on the folder itself. 
- An **image**: use the image itself.
- A **library collection**: use the `lead image`.
- A **page:** use the `lead image`.
- A **profile**: use the image indicated in the `profile image` field.
- A **publication**: use the `lead image`. 

It looks like this will require modifications to code in the [`image_url` function in src/isaw.policy/src/isaw/policy/metadata.py](https://github.com/isawnyu/isaw.web/blob/8219953ef491dc954178506e8bcc6fa5c15ff2b9/src/isaw.policy/src/isaw/policy/metadata.py#L75) and or the code elsewhere in that module that invokes it.

When implementing this, it would likely be a good idea to verify that the other open-graph data that the [`OpenGraphTagViewlet`](https://github.com/isawnyu/isaw.web/blob/8219953ef491dc954178506e8bcc6fa5c15ff2b9/src/isaw.policy/src/isaw/policy/metadata.py#L75) tries to add to the HTML is properly completed and serialized for each of the content types enumerated above.

**Comment by alecpm** [comment]
This is in place on isaw.jazkarta.com for testing.

**Comment by paregorios** [comment]
I have tested on staging.

1. Selection of `lead image` for all content types seems to be working, except for news items, which used to work but now don't ([see latest comment on #472 at bottom](https://github.com/isawnyu/isaw.web/issues/472#issuecomment-1944163768))
2. There seems to be as yet no fallback behavior for when a lead image is not defined (see first sentence, third paragraph beginning "Notes: For any of the below ..." in [the issue description above](https://github.com/isawnyu/isaw.web/issues/457#issue-1505384158).
3. We do not yet seem to be using `og:image:alt` ([see comment above](https://github.com/isawnyu/isaw.web/issues/457#issuecomment-1489028504))

**Comment by paregorios** [comment]
@alecpm 

Here is what I see from my testing:

# 1. Supported content types

This ticket asked for a number of content types to be supported. As I noted [in a comment on #472](https://github.com/isawnyu/isaw.web/issues/472#issuecomment-1944792711): "when a content type supports lead image (or equivalent) and when a lead image has been uploaded for a given object of such a type, og:image uses the desired value".  In other words, all of these are now working, except for `collection`s. Our collections do not currently support assignment of lead images, so they should have been out of scope for this ticket from the beginning. __Nothing to do here.__

# 2. `og:image:alt`

Whenever an alternate text field is defined on an object, this meta property is now properly populated. So there is __nothing else to do here.__ (NB: `profile`s do not have alternative text fields defined, so the absence of `og:image:alt` therein is to be expected.

# 3. fallback selection of image to use for `og:image`.

This does not seem to have been implemented yet. For example:

- [ ] The "[about ISAW](https://isaw.jazkarta.com/about)" `page` has no lead image assigned. But its body text contains embedded images. Per the spec on this ticket, [the first body image](https://isaw.jazkarta.com/home-slides/marta-class1.jpg/@@images/f3750b77-3050-4e30-9dc0-09a77ede9fc1.jpeg) and its associated alt text should be used to populate `og:image` and `og:image:alt` since there is no proper lead image. __This is still a todo.__

**Comment by alecpm** [comment]
@paregorios The implementation of a fallback to an image found within the body text is likely to be error prone and have a negative performance impact on all content without lead images set. If there are common cases where it's not possible to set a lead image, then I can certainly write a fallback method, but I wonder if this is overkill?

Many social sites (Facebook comes to mind) attempt to choose an image from the page automatically if no `og:image` is provided. How well that feature works depends on the structure of the page HTML.

**Comment by paregorios** [comment]
@alecpm given the potential performance hit, let's back off from the fallback method feature for now, splitting it off from this ticket. Therefore, I think there's nothing else to do on this ticket, so once any code review you've requested is done, I would be happy to see associated code deployed to production.

**Comment by paregorios** [comment]
Indeed, Facebook does grab the first embedded image on one of our `page`s without a lead image. Sadly, we often use smaller images than Facebook wants so they end up with an oversampled, upscaled result. But that's still better than the ISAW logo mess we would get before.

Here's a link to [the about page](https://isaw.jazkarta.com/about) as seen in photoshop:

![Screen Shot 2024-02-14 at 4 42 39 PM](https://github.com/isawnyu/isaw.web/assets/263285/faa3bff0-7fe8-45b9-baf6-237ca739e342)

**Comment by alecpm** [comment]
This is live on production.

I recommend adding a Summary/Description on the site homepage and also at `/edit` (which is an alias for the portal site settings. Maybe the same text in the about section of the slider?

> ISAW is a center for advanced scholarly research and graduate education, which aims to encourage particularly the study of the economic, religious, political and cultural connections between ancient civilizations. It offers both doctoral and postdoctoral programs, with the aim of training a new generation of scholars who will enter the global academic community and become intellectual leaders.

**Comment by paregorios** [comment]
This is working well on production.
